framework/src/actions/SplitMeshAction.C

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SplitMeshAction.h"
+
+#include "MooseApp.h"
+#include "MooseUtils.h"
+#include "MooseMesh.h"
+#include "libmesh/checkpoint_io.h"
+
+template <>
+InputParameters
+validParams<SplitMeshAction>()
+{
+  return validParams<Action>();
+}
+
+SplitMeshAction::SplitMeshAction(InputParameters params) : Action(params) {}
+
+void
+SplitMeshAction::act()
+{
+  auto mesh = _app.actionWarehouse().mesh();
+
+  if (mesh->getFileName() == "" && _app.parameters().get<std::string>("split_file") == "")
+    mooseError("Output mesh file name must be specified (with --split-file) when splitting "
+               "non-file-based meshes");
+
+  auto splitstr = _app.parameters().get<std::string>("split_mesh");
+  std::vector<unsigned int> splits;
+  bool success = MooseUtils::tokenizeAndConvert(splitstr, splits, ", ");
+  if (!success)
+    mooseError("invalid argument for --split-file: '", splitstr, "'");
+
+  for (std::size_t i = 0; i < splits.size(); i++)
+  {
+    processor_id_type n = splits[i];
+    Moose::out << "Splitting " << n << " ways..." << std::endl;
+
+    auto cpr = libMesh::split_mesh(*mesh, n);
+    Moose::out << "    - writing " << cpr->current_processor_ids().size() << " files per process..."
+               << std::endl;
+    cpr->binary() = true;
+    auto fname = MooseUtils::stripExtension(mesh->getFileName()) + ".cpr";
+    if (mesh->getFileName() == "")
+      fname = _app.parameters().get<std::string>("split_file");
```

Comment:
There are a few ways to go here.  Line 52 could be before line 50... but that would mean that running the same command-line twice might not work.

Probably better just to error if the `split_file` does not have a `.cpr`.

Reply:
Actually - let's add the `.cpr` here... and then make sure it gets added properly when we're reading in too.

Reply:
It is possible that it is cpa or the user renamed it to something random though.  I think if the user uses the ``--split-file`` flag we should just use that wholesale - this especially makes sense with tab completion.

Reply:
It actually doesn't work though - have you tried it?

If the user names it anything other than `.cpr` it doesn't work.  I think we should make sure that `.cpr` is on the end (if the user passes it just leave it alone - if they don't then append it... for both writing and reading)

Reply:
I'll add the check for the extension when writing.